core/modules/savers/gitlab.js

```diff
@@ -48,12 +48,9 @@ GitLabSaver.prototype.save = function(text,method,callback) {
 	var uri = endpoint + "/projects/" + encodeURIComponent(repo) + "/repository/";
 	// Perform a get request to get the details (inc shas) of files in the same path as our file
 	$tw.utils.httpRequest({
-		url: uri + "tree/" + encodeURIComponent(path.replace(/^\/+|\/$/g, '')),
+		url: uri + "tree/?path=" + encodeURIComponent(path.replace(/^\/+|\/$/g, '')) + "&branch=" + encodeURIComponent(branch.replace(/^\/+|\/$/g, '')),
```

Comment:
We would generally use the `data` property to pass the `path` and `branch` query strings to be assembled by `$tw.utils.httpRequest` (as the previous code did for the `branch` parameter). Is there a reason not to use that approach here?

Reply:
I tried to define the `branch` and` path` parameters in `data`, but it doesn't want to work, only @Janno 's modification works properly: it's a `GET` request, thus you cannot send `data` with it. Although [this bug is present in the GitHub saver](https://github.com/Jermolene/TiddlyWiki5/blob/3732c83ad476fbf8d2c0536da715ed9df456517a/core/modules/savers/github.js#L50-L56) as well, it only contains the `branch` parameter, so we haven't noticed yet (probably everyone uses `master`).